atuin-client/src/database.rs

```diff
@@ -286,27 +287,96 @@ impl Database for Sqlite {
         let query = query.to_string().replace('*', "%"); // allow wildcard char
         let limit = limit.map_or("".to_owned(), |l| format!("limit {}", l));
 
-        let query = match search_mode {
-            SearchMode::Prefix => query,
-            SearchMode::FullText => format!("%{}", query),
-            SearchMode::Fuzzy => query.split("").join("%"),
+        let (query_sql, query_params) = match search_mode {
+            SearchMode::Prefix => ("command like ?1".to_string(), vec![format!("{}%", query)]),
+            SearchMode::FullText => ("command like ?1".to_string(), vec![format!("%{}%", query)]),
+            SearchMode::Fuzzy => {
+                let split_regex = Regex::new(r" +").unwrap();
+                let terms: Vec<&str> = split_regex.split(query.as_str()).collect();
+                let num_terms = terms.len();
+                let mut query_sql = "".to_string();
+                let mut query_params = std::vec::Vec::with_capacity(num_terms);
+                let mut was_or = false;
+                for (i, query_part) in terms.into_iter().enumerate() {
+                    // TODO smart case mode could be made configurable like in fzf
+                    let (operator, glob) = if query_part.contains(char::is_uppercase) {
+                        ("glob", '*')
+                    } else {
+                        ("like", '%')
+                    };
+                    let (is_inverse, query_part) = if query_part.starts_with('!') {
+                        (true, query_part.strip_prefix('!').unwrap())
+                    } else {
+                        (false, query_part)
+                    };
```

Comment:
```suggestion
                    let (is_inverse, query_part) = if let Some(prefix) = query_part.strip_prefix('!') {
                        (true, prefix)
                    } else {
                        (false, query_part)
                    };
```

Reply:
Or maybe even a match statement of similar effect:
```rust
                    let (is_inverse, query_part) = match query_part.strip_prefix('!') {
                        Some(prefix) => (true, prefix),
                        None => (false, query_part),
                    };
```

Reply:
used `match`